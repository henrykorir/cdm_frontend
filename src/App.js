import { useState } from 'react'
import NavBar from './components/NavBar'
import Main from './components/Main'

const App = () => {
	const [ state, setState ] = useState(1)
	const changeState = ( value ) => {
		setState(value)
	}
	return (
		<div className="min-w-11/12 min-h-screen items-center justify-center top-0">
			<NavBar onChangeState = { changeState }/>
			<Main state ={ state }/>
		</div>
	)
}

export default App;
