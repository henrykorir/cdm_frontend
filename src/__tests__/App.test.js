//import { render, screen } from '@testing-library/react';
import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import App from '../App';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('it should match 1', () => {
	act(()=>{
		render(<App />, container);
	});
	expect(container.innerHTML).toMatchSnapshot();
})
