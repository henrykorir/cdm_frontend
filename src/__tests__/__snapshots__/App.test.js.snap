// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match 1 1`] = `"<div class=\\" min-w-11/12 min-h-screen items-center justify-center top-0\\"><div style=\\"border-bottom: .5px solid grey;\\" class=\\"w-full text-center flex justify-center flex-col bg-green-200 pl-1 py-2\\"><ul class=\\"flex space-x-4 content-around text-center items-center text-blue-800\\"><li class=\\"cursor-pointer\\"><a href=\\"/criteria\\">criteria</a></li><li class=\\"cursor-pointer\\"><a href=\\"/patients\\">patients</a></li></ul></div><div>Loading...</div></div>"`;
